sdk/resources/mgmt/src/main/java/com/azure/management/resources/fluentcore/arm/implementation/AzureConfigurableImpl.java

```diff
@@ -92,21 +114,45 @@ public T withProxy(Proxy proxy) {
         return null;
     }
 
-    protected RestClient buildRestClient(AzureTokenCredential credential, AzureEnvironment.Endpoint endpoint) {
-        RestClient client = restClientBuilder
-                .withBaseUrl(credential.getEnvironment(), endpoint)
-                .withCredential(credential)
-//                .withPolicy(new ProviderRegistrationPolicy())
-//                .withPolicy(new ResourceManagerThrottlingPolicy())
-                .buildClient();
-        // TODO: Add proxy support
-//        if (client.httpClient().proxy() != null) {
-//            credentials.withProxy(client.httpClient().proxy());
-//        }
-        return client;
+    @Override
+    public T withScope(String scope) {
+        Objects.requireNonNull(scope);
+        this.scopes.add(scope);
+        return (T) this;
+    }
+
+    @Override
+    public T withHttpClient(HttpClient httpClient) {
+        Objects.requireNonNull(httpClient);
+        this.httpClient = httpClient;
+        return (T) this;
+    }
+
+    @Override
+    public T withConfiguration(Configuration configuration) {
+        Objects.requireNonNull(configuration);
+        this.configuration = configuration;
+        return (T) this;
+    }
+
+    protected HttpPipeline buildHttpPipeline(TokenCredential credential, AzureProfile profile) {
```

Comment:
Does this need to be in sync with `HttpPipelineProvider.buildHttpPipeline(TokenCredential credential, AzureProfile profile)`?

`ProviderRegistrationPolicy` and `ResourceManagerThrottlingPolicy` seems not here?

Reply:
This is a TODO item I marked in provider class. I will discuss with @ChenTanyi for the basic policies we should provide if the users don't make customization.